### PR TITLE
Add searchUsers method

### LINKED ISSUE #829: [REQUEST] searchUsers() function
There is no function to search for users from what I have seen. It would be nice to have and I am surprised it isn't already in noblox.

Endpoint: https://users.roblox.com/v1/users/search

### PR DESCRIPTION
**Pull request by Regalijan**
Closes #829

### CODE REVIEW

File: lib/users/searchUsers.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * ✅ Gets a list of users matching the keyword
+ * @category User
+ * @alias searchUsers
+ * @param {string} keyword - The search term to use
+ * @param {10 | 25 | 50 | 100} limit - The maximum number of matching users to return
+ * @param {string} cursor - The cursor to use when fetching the next or previous page
+ * @returns {Promise<UserSearchResult[]>}
+ * @example const noblox = require("noblox.js")
+ * // Log in using cookie
```
Comment (Neztore) [inline lib/users/searchUsers.js:18]: pedantic, but you are not logging in using the cookie here

File: lib/users/searchUsers.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      keyword
+    }
+  })
+}
+
+exports.func = function (args) {
+  return searchUsers(
+    args.jar,
+    args.keyword,
+    args.limit ?? 10,
```
Comment (Neztore) [inline lib/users/searchUsers.js:39]: Is this some kind of new-fangled syntax? Excuse my ignorance

File: lib/users/searchUsers.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+// Includes
+const getPageResults = require('../util/getPageResults.js').func
+
+// Args
+exports.required = ['keyword']
+exports.optional = ['limit', 'cursor']
```
Comment (Neztore) [inline lib/users/searchUsers.js:6]: Should jar be included as an optional arg here

**Review by Neztore — CHANGES_REQUESTED** [review]

File: lib/users/searchUsers.js (hunk 2 above)
Comment (Regalijan) [reply lib/users/searchUsers.js:39]: Which part specifically?

Comment (Regalijan) [reply lib/users/searchUsers.js:39]: Oh sorry it expanded to show much more than was intended, but yes this is new(ish).

It's node 14 and later so we should be fine here.

**Review by Neztore — APPROVED** [review]
thanks for your changes; ticking

**Pull request merged**